### PR TITLE
Fix StoreResults spec; added required PhysicsGroup argument

### LINKED ISSUE #8075: Reviving StoreResults
For the record, I found this old request:
https://cmsweb.cern.ch/reqmgr2/fetch?rid=jbadillo_StoreResults_115623_v1_150813_113316_6374

which comes from this ggus ticket:
https://ggus.eu/index.php?mode=ticket_info&ticket_id=115623

Then giving this template a shot:
https://github.com/dmwm/WMCore/blob/master/test/data/ReqMgr/requests/DMWM/StoreResults.json

it fails to be injected because of:
* SiteWhitelist cannot be set during creation (remove it)
* ConfigCacheID was not provided (when it shouldn't be mandatory for this case)
* something else I missed...

Not a problem, but it seems `CmsPath` is just a legacy stuff.

PhysicsGroup seems to be part of the MergedLFNBase parameter. I think we could create a `PhysicsGroup` parameter and then lower + append it to the MergedLFNBase (which should always be "/store/results" btw). @ericvaandering @ticoann what do you think?

PS.: It seems the information we should be expecting from requestors are:
```
ScramArch
CMSSWVersion
DbsUrl
InputDataset
PhysicsGroup (which is appended to MergedLFNBase)
SiteWhitelist (where the input is and where the output will be written to)
```

### PR DESCRIPTION
**Pull request by amaltaro**
Fixes #8075 
As you can see, we update the MergedLFNBase inside the spec such that it reflects the PhysicsGroup the user is member of. I'm not sure there is a store results namespace after each of those physics groups? @ericvaandering would you know?

### CODE REVIEW

**Comment by amaltaro** [comment]
So, my question is, should we automatically map a PhysicsGroup to a MergedLFNBase? E.g.
PhysicsGroup: HIN
MergedLFNBase has to be '/store/results/hin' instead of '/store/results'

Otherwise, we can leave the default value '/store/results' and let P&R team, or whoever creates/assigns these workflows to properly find out and set the MergedLFNBase.

Maybe @thongonary can comment on it (please add Matteo too, I have no idea what's his GH user)

**Comment by mcremone** [comment]
Thinking twice about it I am not sure it is what we really want. It looks to us that:

- not all the physics groups appear there
- it is not clear if we should treat elevated datasets differently from other datasets that get /store/result by default

**Comment by amaltaro** [comment]
I did one more minor change that isn't tested, but it should be good to go to further integration tests @ticoann

**Review by ticoann — APPROVED** [review]

File: src/python/WMCore/WMSpec/StdSpecs/StoreResults.py [hunk 1]
```diff
@@ -92,29 +94,29 @@ def __call__(self, workloadName, arguments):
...
-                    "RunWhitelist" : {"default" : [], "type" : makeList,
-                                      "optional" : True, "validate" : lambda x: all([int(y) > 0 for y in x]),
-                                      "attr" : "runWhitelist", "null" : False}}
+        specArgs = {"RequestType": {"default": "StoreResults", "optional": False},
+                    "InputDataset": {"optional": False, "validate": dataset, "null": False},
+                    "ConfigCacheID": {"optional": True, "null": True},
+                    "DataTier": {"default": "USER", "type": str,
+                                 "optional": True, "validate": None,
+                                 "attr": "dataTier", "null": False},
+                    "PhysicsGroup": {"default": "", "optional": False,
```
[the remark this answers is not in the document]
Comment (amaltaro) [reply src/python/WMCore/WMSpec/StdSpecs/StoreResults.py]: No default value means default it to None :) The default empty string will fail validation.
But I agree that None would have the same result, if not more clear.

**Comment by amaltaro** [comment]
@ticoann @ericvaandering @mcremone @thongonary I've also just made this wiki
https://github.com/dmwm/WMCore/wiki/StoreResults-requests

feel free to correct/suggest any improvements.

**Comment by ericvaandering** [comment]
I made a few minor ones. Thanks, Alan.

**Pull request merged**